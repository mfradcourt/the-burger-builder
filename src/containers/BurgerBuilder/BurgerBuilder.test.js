import React from 'react';
import { shallow, mount } from "enzyme";

import { BurgerBuilder } from './BurgerBuilder';

describe('<BurgerBuilder />', () => {
  let wrapper;

  it('should render BurgerBuilder', () => {
    wrapper = shallow(
      <BurgerBuilder/>
    );

    expect(wrapper).toMatchSnapshot();
    wrapper.instance().setState({loading: true});
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().setState({loading: false, error: true});
    expect(wrapper).toMatchSnapshot();
  });

  it('addIngredientHandler', () => {
    wrapper = shallow(
      <BurgerBuilder/>
    );
    wrapper.instance().setState({totalPrice: 4, ingredients:{}});
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().addIngredientHandler('meat');
    expect(wrapper.instance().state.totalPrice).toBe(5.3);
    expect(wrapper).toMatchSnapshot();
  });

  it('removeIngredientHandler', () => {
    wrapper = shallow(
      <BurgerBuilder/>
    );
    wrapper.instance().setState({totalPrice: 4.4, ingredients:{cheese:1}});
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().removeIngredientHandler('cheese');
    expect(wrapper.instance().state.totalPrice).toBe(4);
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().removeIngredientHandler('cheese');
    expect(wrapper.instance().state.totalPrice).toBe(4);
    expect(wrapper).toMatchSnapshot();
  });

});
