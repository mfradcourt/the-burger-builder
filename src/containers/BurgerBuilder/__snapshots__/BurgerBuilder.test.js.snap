// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerBuilder /> addIngredientHandler 1`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={Object {}}
      price={4}
      purchaseCancelled={[Function]}
      purchaseContinued={[Function]}
    />
  </Modal>
  <burger
    ingredients={Object {}}
  />
  <buildControls
    disabled={Object {}}
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={4}
    purchasable={false}
  />
</Fragment>
`;

exports[`<BurgerBuilder /> addIngredientHandler 2`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={
        Object {
          "meat": NaN,
        }
      }
      price={5.3}
      purchaseCancelled={[Function]}
      purchaseContinued={[Function]}
    />
  </Modal>
  <burger
    ingredients={
      Object {
        "meat": NaN,
      }
    }
  />
  <buildControls
    disabled={
      Object {
        "meat": false,
      }Â²
    }
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={5.3}
    purchasable={false}
  />
</Fragment>
`;

exports[`<BurgerBuilder /> removeIngredientHandler 1`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={
        Object {
          "cheese": 1,
        }
      }
      price={4.4}
      purchaseCancelled={[Function]}
      purchaseContinued={[Function]}
    />
  </Modal>
  <burger
    ingredients={
      Object {
        "cheese": 1,
      }
    }
  />
  <buildControls
    disabled={
      Object {
        "cheese": false,
      }
    }
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={4.4}
    purchasable={false}
  />
</Fragment>
`;

exports[`<BurgerBuilder /> removeIngredientHandler 2`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={
        Object {
          "cheese": 0,
        }
      }
      price={4}
      purchaseCancelled={[Function]}
      purchaseContinued={[Function]}
    />
  </Modal>
  <burger
    ingredients={
      Object {
        "cheese": 0,
      }
    }
  />
  <buildControls
    disabled={
      Object {
        "cheese": true,
      }
    }
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={4}
    purchasable={false}
  />
</Fragment>
`;

exports[`<BurgerBuilder /> removeIngredientHandler 3`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  >
    <OrderSummary
      ingredients={
        Object {
          "cheese": 0,
        }
      }
      price={4}
      purchaseCancelled={[Function]}
      purchaseContinued={[Function]}
    />
  </Modal>
  <burger
    ingredients={
      Object {
        "cheese": 0,
      }
    }
  />
  <buildControls
    disabled={
      Object {
        "cheese": true,
      }
    }
    ingredientAdded={[Function]}
    ingredientRemoved={[Function]}
    ordered={[Function]}
    price={4}
    purchasable={false}
  />
</Fragment>
`;

exports[`<BurgerBuilder /> should render BurgerBuilder 1`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  />
  <spinner />
</Fragment>
`;

exports[`<BurgerBuilder /> should render BurgerBuilder 2`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  >
    <spinner />
  </Modal>
  <spinner />
</Fragment>
`;

exports[`<BurgerBuilder /> should render BurgerBuilder 3`] = `
<Fragment>
  <Modal
    modalClosed={[Function]}
    show={false}
  />
  <p>
    Ingredients can't be loaded!
  </p>
</Fragment>
`;
